### DISCUSSION TITLE
Is it possible to create an index field on an array type?

### DISCUSSION THREAD

**Discussion by kiranchitturi** [milvus-io/milvus#34939] [Q&A and General discussion]
I am creating an array field of varchar type and want to create an index on it for faster searches but it doesn't seem to work. Is this a supported feature right now?

**Comment by xiaofan-luan** [comment]
I think you can created inverted index on array.
is it on 2.4 branch already @longjiquan

**Reply by zhu3359** [reply]
is there a demo script?

**Reply by yhmo** [reply]
@zhu3359 
https://milvus.io/docs/index-scalar-fields.md
```
index_params = MilvusClient.prepare_index_params() #  Prepare an IndexParams object

index_params.add_index(
    field_name="scalar_2", # Name of the scalar field to be indexed
    index_type="INVERTED", # Type of index to be created
    index_name="inverted_index" # Name of the index to be created
)

client.create_index(
  collection_name="test_scalar_index", # Specify the collection name
  index_params=index_params
)
```

**Reply by zhu3359** [reply]
@yhmo building inverted index for a array field means building inverted index for every single element in the array, is that right?

**Reply by yhmo** [reply]
Yes, I think so.

**Comment by kiranchitturi** [comment]
@xiaofan-luan Is it in 2.4.4? Can you send me a sample command? I could not find documentation on it

What should be the index_type? Trying something like fails for an array of varchars

```
Collection("test_index_array").create_index("array", index_type="INVERTED", index_name="array_index")
```
